account_banking_mandate/tests/test_invoice_mandate.py

```diff
@@ -0,0 +1,318 @@
+
+        with self.assertRaises(UserError):
+            invoice.mandate_id = mandate_2
+
+    def _create_res_partner(self, name):
+        return self.env["res.partner"].create({"name": name})
+
+    def _create_res_bank(self, name, bic, city, country):
+        return self.env["res.bank"].create(
+            {"name": name, "bic": bic, "city": city, "country": country.id}
+        )
+
+    def setUp(self):
+        res = super(TestInvoiceMandate, self).setUp()
+        self.company = self.env.ref("base.main_company")
+
+        self.partner = self._create_res_partner("Peter with ACME Bank")
+        self.acme_bank = self._create_res_bank(
+            "ACME Bank", "GEBABEBB03B", "Charleroi", self.env.ref("base.be")
+        )
+
+        bank_account = self.env["res.partner.bank"].create(
+            {
+                "acc_number": "0023032234211123",
+                "partner_id": self.partner.id,
+                "bank_id": self.acme_bank.id,
+                "company_id": self.company.id,
+            }
+        )
+
+        self.company_2 = self.env["res.company"].create({"name": "Company 2"})
+
+        self.mandate = self.env["account.banking.mandate"].create(
+            {
+                "partner_bank_id": bank_account.id,
+                "signature_date": "2015-01-01",
+                "company_id": self.company.id,
+            }
+        )
+
+        self.mandate.validate()
+
+        self.mode_inbound_acme = self.env["account.payment.mode"].create(
+            {
+                "name": "Inbound Credit ACME Bank",
+                "company_id": self.company.id,
+                "bank_account_link": "variable",
+                "payment_method_id": self.env.ref(
+                    "account.account_payment_method_manual_in"
+                ).id,
+            }
+        )
+        bank_journal = self.env["account.journal"].search(
+            [
+                ("type", "=", "bank"),
+                ("company_id", "=", self.company.id),
+            ],
+            limit=1,
+        )
+        self.mode_inbound_acme.variable_journal_ids = bank_journal
+        self.mode_inbound_acme.payment_method_id.mandate_required = True
+        self.mode_inbound_acme.payment_order_ok = True
+
+        self.partner.customer_payment_mode_id = self.mode_inbound_acme
+
+        self.invoice_account = self.env["account.account"].search(
+            [
+                (
+                    "user_type_id",
+                    "=",
+                    self.env.ref("account.data_account_type_receivable").id,
+                ),
+                ("company_id", "=", self.company.id),
+            ],
+            limit=1,
+        )
+        invoice_line_account = (
+            self.env["account.account"]
+            .search(
+                [
+                    (
+                        "user_type_id",
+                        "=",
+                        self.env.ref("account.data_account_type_expenses").id,
+                    ),
```

Comment:
The field `user_type_id` no longer exists on 16.0. You should change it to the selection field `account_type`